From d9c42164317cbb5cc4c09ce71b612b5fb1dca09d Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 26 Aug 2014 15:30:24 -0400
Subject: [ethernet] igbvf: Miscellaneous conversions to ETH_ALEN

Message-id: <1409067034-17415-4-git-send-email-sassmann@redhat.com>
Patchwork-id: 89214
O-Subject: [RHEL7.1 PATCH 03/13] net:drivers/net: Miscellaneous conversions to ETH_ALEN
Bugzilla: 1091122
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

From: Joe Perches <joe@perches.com>

Convert the memset/memcpy uses of 6 to ETH_ALEN
where appropriate.

Also convert some struct definitions and u8 array
declarations of [6] to ETH_ALEN.

Signed-off-by: Joe Perches <joe@perches.com>
Acked-by: Arend van Spriel <arend@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit d458cdf712e0c671e8e819abb16ecd6e44f9daec)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/igbvf/vf.c b/drivers/net/ethernet/intel/igbvf/vf.c
index eea0e10..955ad8c 100644
--- a/drivers/net/ethernet/intel/igbvf/vf.c
+++ b/drivers/net/ethernet/intel/igbvf/vf.c
@@ -154,7 +154,7 @@ static s32 e1000_reset_hw_vf(struct e1000_hw *hw)
   ret_val = mbx->ops.read_posted(hw, msgbuf, 3);
   if (!ret_val) {
    if (msgbuf[0] == (E1000_VF_RESET | E1000_VT_MSGTYPE_ACK))
-    memcpy(hw->mac.perm_addr, addr, 6);
+    memcpy(hw->mac.perm_addr, addr, ETH_ALEN);
    else
     ret_val = -E1000_ERR_MAC_INIT;
   }
@@ -314,7 +314,7 @@ static void e1000_rar_set_vf(struct e1000_hw *hw, u8 * addr, u32 index)
 
  memset(msgbuf, 0, 12);
  msgbuf[0] = E1000_VF_SET_MAC_ADDR;
- memcpy(msg_addr, addr, 6);
+ memcpy(msg_addr, addr, ETH_ALEN);
  ret_val = mbx->ops.write_posted(hw, msgbuf, 3);
 
  if (!ret_val)
-- 
1.7.1